internal/command/init.go

```diff
@@ -761,6 +761,20 @@ func (c *InitCommand) getProviders(config *configs.Config, state *states.State,
 				// but rather just emit a single general message about it at
 				// the end, by checking ctx.Err().
 
+			case providercache.ErrProviderChecksumMiss:
+				// This is a special kind of error that can often be fixed using
+				// the `terraform providers lock` command. We're just going to
+				// amend the actual error message with some extra information
+				// about how to fix this.
+
+				diags = diags.Append(tfdiags.Sourceless(
+					tfdiags.Error,
+					"Failed to install provider",
+					fmt.Sprintf("Error while installing %s v%s: %s\n\nYou can ensure the current platform, %s, is included in the dependency lock file by running: `terraform providers lock -provider=%s`.\n\nIf this does not fix the problem you may need to reset any provider caching present in your setup or make sure you are connecting to valid provider distributions.",
```

Comment:
I think we might need to be careful about making this suggestion without some more context about the implications.

Specifically, when combining with this with the other change we recently made so that `terraform providers lock` will not honor existing checksums, I think this suggestion amounts to "The checksums didn't match the lock file, which you should fix by discarding the checksums in the lock file".

I must admit I'm not sure how to inject further subtlety here without making the message incredibly verbose. I think the main qualifier we want to get across here is "If you got into this situation by not having a complete set of checksums for all platforms you intend to use...", but I don't really know how to concisely explain to a user how they would determine if that predicate is true, and how to securely differentiate it from the bad case where someone has actually maliciously tampered with the plugin package. :thinking: 

I think we might need to ask the product security team for a second opinion on whatever we decide here too, since we're changing the characteristics of a pretty sensitive area for the threat models of some users.

Reply:
I've changed this so we just link directly to the relevant documentation, hopefully that avoids any potential security problems while still giving users a jump start on the potential fixes.

The relevant documentation does concretely suggest `terraform providers lock`, while providing context around how this could have happened.